### PR TITLE
email test infrastructure

### PR DESCRIPTION
**Pull request by individual-it**
## Description
infrastructure to be able to write acceptance tests for features that send emails
main changes:
1. the `mailhogUrl` from the behat.yml was not used at all, now we are taking that setting from env. variables what makes it possible to easily adapt the URL for CI
2. the `MailTool` trait went into an own context and Helper

## Related Issue
https://github.com/owncloud/QA/issues/537

## Motivation and Context
be able to test email sending functionality

## How Has This Been Tested?
written tests for it in the notifications app https://github.com/owncloud/notifications/pull/177

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/TestHelpers/EmailHelper.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		$response = $client->send($request);
+
+		$json = json_decode($response->getBody()->getContents());
+		return $json;
+	}
+
+	/**
+	 * 
+	 * @param string $mailhogUrl
+	 * @return ResponseInterface
```
Comment (phil-davis) [inline tests/TestHelpers/EmailHelper.php]: param and return have an empty comment line between them in our current phpdoc "standard"

File: tests/acceptance/features/bootstrap/EmailContext.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+use Behat\Behat\Context\Context;
+use Behat\Behat\Context\SnippetAcceptingContext;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use Behat\Gherkin\Node\PyStringNode;
+use TestHelpers\EmailHelper;
+
+require_once 'bootstrap.php';
+
+/**
+ * context file for email relates steps.
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/EmailContext.php]: s/relates/related

**Review by phil-davis — COMMENTED** [review]
Looks good. Just little comment things.

File: tests/TestHelpers/EmailHelper.php (hunk 1 above)
Comment (individual-it) [reply tests/TestHelpers/EmailHelper.php]: why didn't phpcs shout at me, is that rule not in our phpcs.xml ?

File: tests/acceptance/features/bootstrap/EmailContext.php (hunk 2 above)
Comment (individual-it) [reply tests/acceptance/features/bootstrap/EmailContext.php]: fixed

File: tests/TestHelpers/EmailHelper.php (hunk 1 above)
Comment (phil-davis) [reply tests/TestHelpers/EmailHelper.php]: If you add a description to the PHPdoc then it shouts:
```
	/**
	 * short description here
	 * 
	 * @param string $mailhogUrl
	 * @return ResponseInterface
	 */
```
and you get:
```
phpcs --standard=phpcs.xml tests/TestHelpers/EmailHelper.php 
E 1 / 1 (100%)



FILE: /home/phil/ownCloud/core/tests/TestHelpers/EmailHelper.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------------
 56 | ERROR | Tag cannot be grouped with parameter tags in a doc comment (Generic.Commenting.DocComment.NonParamGroup)
-----------------------------------------------------------------------------------------------------------------------

Time: 39ms; Memory: 6Mb
```
so ``phpcs`` is not parsing the tag blocks properly when there is no "short decription".

And we have turned off the requirement for a shirt description, so it does not complain about that.

**Review by phil-davis — APPROVED** [review]

File: tests/TestHelpers/EmailHelper.php (hunk 1 above)
Comment (phil-davis) [reply tests/TestHelpers/EmailHelper.php]: And ``phpcs` fix merged here, for ``phpcs`` 3.3.0 release:
https://github.com/squizlabs/PHP_CodeSniffer/pull/1990

**Comment by phil-davis** [comment]
Backport ``stable10`` #31112

**Pull request merged**